backend/kesaseteli/applications/tests/test_excel_export.py

```diff
@@ -257,6 +282,110 @@ def employer_summer_voucher_sorting_key(voucher: EmployerSummerVoucher):
+        + [
+            YouthApplicationFactory(status=status)
+            for status in YouthApplicationStatus.active_values()
+        ]
+        + [
+            ActiveVtjTestCaseYouthApplicationFactory(last_name=vtj_test_case)
+            for vtj_test_case in VtjTestCase.values
+        ]
+    )
+    # Update created_at specially as its initial save uses auto_now_add
+    for app in apps:
+        app.created_at -= timedelta(seconds=random.randint(0, 60 * 60 * 24 * 10))
+        app.save(update_fields=["created_at"])
+        app.refresh_from_db()
+    assert min(app.created_at for app in apps) != max(app.created_at for app in apps)
+    apps = sorted(apps, key=youth_application_sorting_key)
+
+    with translation.override("en"):  # Should still use Finnish translations
+        response = staff_client.get(youth_excel_download_url())
+
+    assert isinstance(response, StreamingHttpResponse)
+
+    workbook = openpyxl.load_workbook(filename=BytesIO(response.getvalue()))
+    active_worksheet = workbook.active
+
+    rows_generator = active_worksheet.rows
+    header_row = next(rows_generator)
+    data_rows = [next(rows_generator) for _ in apps]
+    with pytest.raises(StopIteration):
+        next(rows_generator)
+
+    output_column_names = tuple(column.value for column in header_row)
+    with translation.override("fi"):  # Make extra sure to test for Finnish column names
+        expected_output_columns_names = (
+            YouthApplicationExcelExportView.output_column_names()
+        )
+    assert output_column_names == expected_output_columns_names
+
+    source_fields = YouthApplicationExcelExportView.source_fields()
+    for data_row, app in zip(data_rows, apps):
+        for output_cell, source_field in zip(data_row, source_fields):
+            output_value = output_cell.value
+            if source_field == "application_date":
+                assert date.fromisoformat(output_value) == localdate(app.created_at)
+            elif source_field == "id":
+                assert output_value == str(app.id)
+            elif source_field.endswith("application_year"):
+                assert int(output_value) == localdate(app.created_at).year
+            elif source_field == "summer_voucher_serial_number":
+                if output_value is None:
+                    assert not app.has_youth_summer_voucher
+                else:
+                    assert (
+                        int(output_value)
+                        == app.youth_summer_voucher.summer_voucher_serial_number
+                    )
+            elif source_field == "birth_year":
+                assert int(output_value) == app.birthdate.year
+            elif source_field == "birthdate":
+                assert date.fromisoformat(output_value) == app.birthdate
+            elif source_field == "is_unlisted_school":
+                assert output_value in ["Kyllä", "Ei"]
+                assert output_value != "Ei" or not app.is_unlisted_school
```

Comment:
double negations, without negations would be more readable?
`output_value == "Kyllä" or app.is_unlisted_school`

Reply:
I was trying to use logical implication here i.e. a ⇒ b <=> !a or b:
```
assert output_value != "Ei" or not app.is_unlisted_school
assert output_value != "Kyllä" or app.is_unlisted_school
```
If Python had logical implication operator I could have written:
```
assert (output_value == "Ei") ⇒ not app.is_unlisted_school
assert (output_value == "Kyllä") ⇒ app.is_unlisted_school
```

But I can actually combine the two cases into a more concise and clearer assertion because we've checked already that output_value must be either "Kyllä" or "Ei":
```
assert (output_value == "Kyllä") == app.is_unlisted_school
```

Reply:
Combined the asserts to a single one as above